pkg/distributor/otel.go

```diff
@@ -203,7 +213,7 @@ func otelMetricsToMetadata(md pmetric.Metrics) []*mimirpb.MetricMetadata {
 				metric := scopeMetrics.Metrics().At(k)
 				entry := mimirpb.MetricMetadata{
 					Type:             otelMetricTypeToMimirMetricType(metric),
-					MetricFamilyName: prometheustranslator.BuildCompliantName(metric, "", true), // TODO expose addMetricSuffixes in configuration (https://github.com/grafana/mimir/issues/5967)
+					MetricFamilyName: prometheustranslator.BuildCompliantName(metric, "", addSuffixes),
```

Comment:
So, when setting the addSuffixes to true or false it needs to be set here too
https://github.com/grafana/mimir/blob/main/pkg/distributor/otel.go#L220
The `prometheus.Settings{}` passed there has a bool for it too.
```
type Settings struct {
	Namespace           string
	ExternalLabels      map[string]string
	DisableTargetInfo   bool
	ExportCreatedMetric bool
	AddMetricSuffixes   bool
}
```
If both are not set to true or false, the metadata won't show up properly in Grafana, becaus the metric name in TSDB won't match the metric name in Metadata storage if the suffix is injected or not injected in both places.

Reply:
Thank you!

Reply:
Fixed.